docs/assets.rst

```diff
@@ -0,0 +1,673 @@
+functions, such as ``gettext()``, the common alias ``_()``, and ``ngettext()``.
+It also parses longer strings in the template tag ``trans``.
+
+Next the changes are merged into the existing Portable Object (``.po``) files,
+such as ``locale/fr/LC_MESSAGES/django.po``, to add new strings and comment out
+removed strings.
+
+Extracting and merging is done with ``make localeextract``, usually
+during :doc:`deployment </deploy>`, when UI strings change.  The changes are
+pushed as a new commit to the mdn-l10n_ repository.
+
+Pontoon detects that the repository has changed, and notifies localization
+teams that there are new strings. In about 48 hours, the most active teams will
+translate strings into the top 10 MDN languages. These are applied by updating
+the ``locale`` submodule during the :doc:`deployment process </deploy>`.
+
+At run time, Machine Object (``.mo``) files, such as
+``locale/fr/LC_MESSAGES/django.mo``, are used by gettext functions, like
+``gettext()`` and ``_()``, to display the localized strings.
+These are built with ``make localecompile`` when creating the production
+images or when a developer wants to see updated translations.
+
+.. _Pontoon: https://pontoon.mozilla.org/projects/mdn/
+.. _mdn-l10n: https://github.com/mozilla-l10n/mdn-l10n
+.. _`git submodule`: https://github.blog/2016-02-01-working-with-submodules/
+.. _gettext: https://en.wikipedia.org/wiki/Gettext
+.. _Puente: https://puente.readthedocs.io/en/latest/
+
+.. _build-locale-js:
+
+Building localization JavaScript
+================================
+Django includes a `JavaScriptCatalog view`_ that provides JavaScript
+implementations of gettext_ functions, as well as translations for each
+locale. It is ineffecient to use this view directly, since it is generated
+on access. For efficiency, django-statici18n_ generates files for each locale
+from the ``JavaScriptCatalog`` output, so they can be served as static assets.
+
+The translation catalog files are created with ``make compilejsi18n``
+from the locale Machine Object ``.mo`` files.  Kuma sets ``STATICI18N_ROOT``
+to ``build/locale``, and the output files have names like
+``build/locale/jsi18n/de/javascript.js``.
+
+.. _`JavaScriptCatalog view`: https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#module-django.views.i18n
+.. _django-statici18n: https://django-statici18n.readthedocs.io/en/latest/
+
+.. _build-ckeditor:
+
+Building CKEditor
+=================
+CKEditor is a complex JavaScript application that provides a WYSIWYG editor
+for MDN wiki pages. It is packaged with plugins, some from third parties,
+and some custom to MDN.
+
+The CKEditor build process is documented on the
+:doc:`CKEditor document</ckeditor>`. The built files are checked into the
+Kuma repository.
+
+.. _include-js-libraries:
+
+Including JS libraries
+======================
+Third-party JavaScript libraries are included in the Kuma repository.
+Some were added manually, and others with Bower_. See
```

Comment:
But the Bower ones are still checked in, right? (I don't remember running any kind of Bower step when setting up my repo...)

Reply:
Yes, Bower was used to download the versions specified in https://github.com/mozilla/kuma/blob/master/bower.json, which were then checked in. It takes a while for Mozilla to trust package managers.